### ISSUE TITLE
Not returning location when location method is "Device only"

### ISSUE THREAD

**Issue by hortemo** [MustansirZia/react-native-fused-location#14]
Hi! I am not able to get a location when using:
- getFusedLocation(true)
- FusedLocation.Constants.BALANCED
- ACCESS_FINE_LOCATION
- Location method "Device only"

This is on a real device for which i.e. Google Maps works fine. It works fine for other location methods (High Accuracy and Battery Saving).

Is this your experience as well, or is it probably something on my part?

**Comment by MustansirZia** [comment]
Hi! 
Does it go inside catch statement or the promise just never resolves?

**Comment by MustansirZia** [comment]
Also, what do you mean by `Location method "Device only"`?

**Comment by hoboman313** [comment]
Well Android has 3 Location modes that you can set: High accuracy, Battery saving and device only. He must be referring to the later.

**Comment by hortemo** [comment]
Yes, I am talking about the location mode in Android settings. I get no error, it just never returns any location data.

**Comment by MustansirZia** [comment]
Hi @hortemo! A more likely cause for this problem is that since you're using `Device only` and forcing a new location update your device is not able to get a location as your device maybe somewhere indoors or has a poor internet connection. You've optimally set the mode to `BALANCED` but externally forced your device to get a new location from your GPS only. I think I should probably implement a timeout feature just so these cases are taken care of.

**Comment by Neha1490** [comment]
@MustansirZia  - Hi, I am also facing the same issue.
I have tried to change Location Priority as HIGH_ACCURACY but still getting the same issue.

**Comment by sraka1** [comment]
@MustansirZia the problem seems to be that if the device setting is set to "Device Only" (GPS only), trying to get the location with "balanced" accuracy never works (irregardless of GPS signal strength). This can be replicated easily in the simulator. A timeout is a good idea. Reading this setting and acting accordingly would be even better.

**Comment by MustansirZia** [comment]
@sraka1 You're probably right, it's about time we make use of a more appropriate callback. `LocationCallback ` does seem a better way to go than the original `LocationCallback `. This requires attention. 
I still, however, want to know why the "balanced" accuracy fails with "Device Only", that also requires some attention.

**Comment by MustansirZia** [comment]
Package updated to `0.2.1` 8f64dbb5922d320bc048d824ba799eb8411c467c  which makes use of `[LocationCallback](https://developers.google.com/android/reference/com/google/android/gms/location/LocationCallback)` instead of `LocationListener`. This would guarantee that the promise would get rejected if no suitable location could be returned (Fresh or old). As @sraka1 had pointed out earlier, LocationListener had no way of knowing that the request had failed inside it so a LocationCallback was a good way to go.
This should also eliminate the need for a timeout.

**Comment by ingvardm** [comment]
actually just saw this bug. If no one else is working on it, ill try to find some time in the next few days and look at it.

**Comment by MustansirZia** [comment]
It didn't go inside catch statement? Which version of the library are you using?
